week2/README.md

```diff
@@ -0,0 +1,88 @@
+
+h) How can you view your App in Light and Dark Modes simultaneously?</br>
+
+**In the Canvas, make sure your preview is running, then click the Variants button and select Color Scheme variants. You can then see your app in Light and Dark modes at the same time, click into one to see a larger view, and click the side arrows to navigate between the two modes.**
+
+i) Below is an image of the Canvas from Xcode. The Canvas is in selectable mode. Can you explain why the red background does not cover the entire button area?</br>
+
+![](https://lh7-us.googleusercontent.com/Rw5e_3ja5JkNz8Ih6v3ZJdKxu5PWB7Gn_Rw5Yl29HJbrzmqQwqt4ozZaEsToqJmb3lGbw9P7NJd0Iw614Mi8cu_41RAJUwem3yo1rn5TQJ33Ol2-l_R2AQz3otScEUTjiQJd7zQe8kSAdTg0)
+
+**Every time you add a modifier to a view, it creates a new view with that modifier applied. So if you add a red `background` modifier before adding `padding`, it will put the red `background` only over the original frame of the button (without the `padding`), and then the `padding` will be added around the `background`. If you switch the modifier order so that `padding` comes first, then the red `background` should cover the entire button area.**
+
+j) Modifier padding(10) adds padding to the view's top, bottom, left, and right sides. How could a padding of 10 be added to only the left and right sides of the view? The answer for this question should be a short section of code.</br>
+
+`.padding(.horizontal, 10)`
+
+k) Provide two reasons why you would want to extract views.
+- **In order to separate out views that you plan on reusing in multiple places. If you're going to use the same type of text view in several places in your app, you can extract that view to a separate named struct so that you can refer to it wherever you want, plugging in different parameters as needed.**
+- **In order to keep your main view clean and easy to navigate. If all your views are in one giant ContentView, your code can easily become bulky and difficult to navigate; extracting views that serve specific functions out to their own structs or even their own files can help you organize your code in a way that's easy to decipher and traverse.**
+
+l) How can you determine, in code, if the device is in Portrait or Landscape mode?</br>
+
+**By declaring `@Enviroment(\.verticalSizeClass)` and `@Enviroment(\.horizontalSizeClass)` variables in your struct, then testing for them in your view using if statements. On an iPhone, if `verticalSizeClass` is regular and `@Enviroment(\.horizontalSizeClass)` is compact, then you're in Portrait mode.**
+
+m) What is a literal value? **The assigned or calculated value of a constant or variable, i.e. the number of an `Int,` or the actual characters inside a `String.`**
+
+o) What are the safe areas?</br>
```

Comment:
Safe areas are layout guides that take into account interface elements like the iPhone's notch or the home indicator. They ensure content is displayed within a visible area and isn't obscured by these interface elements. In SwiftUI, you often use safe areas to prevent placing UI elements where they might be overlapped by system UI or hardware features. This ensures your app's content is always visible and accessible to the user.